samples/net/sockets/http_client/src/main.c

```diff
@@ -0,0 +1,351 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(net_http_client_sample, LOG_LEVEL_DBG);
+
+#include <net/net_ip.h>
+#include <net/socket.h>
+#include <net/tls_credentials.h>
+#include <net/http_client.h>
+
+#include "ca_certificate.h"
+
+#define HTTP_PORT 8000
+#define HTTPS_PORT 4443
+
+#if defined(CONFIG_NET_CONFIG_PEER_IPV6_ADDR)
+#define SERVER_ADDR6  CONFIG_NET_CONFIG_PEER_IPV6_ADDR
+#else
+#define SERVER_ADDR6 ""
+#endif
+
+#if defined(CONFIG_NET_CONFIG_PEER_IPV4_ADDR)
+#define SERVER_ADDR4  CONFIG_NET_CONFIG_PEER_IPV4_ADDR
+#else
+#define SERVER_ADDR4 ""
+#endif
+
+#define MAX_RECV_BUF_LEN 512
+
+static u8_t recv_buf_ipv4[MAX_RECV_BUF_LEN];
+static u8_t recv_buf_ipv6[MAX_RECV_BUF_LEN];
+
+static int setup_socket(sa_family_t family, const char *server, int port,
+			int *sock, struct sockaddr *addr, socklen_t addr_len)
+{
+	const char *family_str = family == AF_INET ? "IPv4" : "IPv6";
+	int ret = 0;
+
+	memset(addr, 0, addr_len);
+
+	if (family == AF_INET) {
+		net_sin(addr)->sin_family = AF_INET;
+		net_sin(addr)->sin_port = htons(port);
+		inet_pton(family, server, &net_sin(addr)->sin_addr);
+	} else {
+		net_sin6(addr)->sin6_family = AF_INET6;
+		net_sin6(addr)->sin6_port = htons(port);
+		inet_pton(family, server, &net_sin6(addr)->sin6_addr);
+	}
+
+	if (IS_ENABLED(CONFIG_NET_SOCKETS_SOCKOPT_TLS)) {
+		sec_tag_t sec_tag_list[] = {
+			CA_CERTIFICATE_TAG,
+		};
+
+		*sock = socket(family, SOCK_STREAM, IPPROTO_TLS_1_2);
+		if (*sock >= 0) {
+			ret = setsockopt(*sock, SOL_TLS, TLS_SEC_TAG_LIST,
+					 sec_tag_list, sizeof(sec_tag_list));
+			if (ret < 0) {
+				LOG_ERR("Failed to set %s secure option (%d)",
+					family_str, -errno);
+				ret = -errno;
+			}
+
+			ret = setsockopt(*sock, SOL_TLS, TLS_HOSTNAME,
+					 TLS_PEER_HOSTNAME,
+					 sizeof(TLS_PEER_HOSTNAME));
+			if (ret < 0) {
+				LOG_ERR("Failed to set %s TLS_HOSTNAME "
+					"option (%d)", family_str, -errno);
+				ret = -errno;
+			}
+		}
+	} else {
+		*sock = socket(family, SOCK_STREAM, IPPROTO_TCP);
+	}
+
+	if (*sock < 0) {
+		LOG_ERR("Failed to create %s HTTP socket (%d)", family_str,
+			-errno);
+	}
+
+	return ret;
+}
+
+static int payload_cb(int sock, struct http_request *req, void *user_data)
+{
+	const char *content[] = {
+		"foobar",
+		"chuncked",
```

Comment:
typo, will fix it, anyway it is part of the HTTP payload so it does not really matter what combination of chars we send

Reply:
> anyway it is part of the HTTP payload so it does not really matter what combination of chars we send

That's true, but still, unless typo is made on purpose, would be nice to keep the code grammar-correct (or otherwise it may lead to a feeling of sloppiness) ;-).